MdeModulePkg/Core/PiSmmCore/Page.c

```diff
@@ -165,7 +165,10 @@ CoreFreeMemoryMapStack (
     // Deque an memory map entry from mFreeMemoryMapEntryList
     //
     Entry = AllocateMemoryMapEntry ();
-    ASSERT (Entry);
+    // MU_CHANGE - Verify that Entry is non-null
+    if (Entry == NULL) {
+      ASSERT (Entry);

```

Comment:
yes, it should return if there is a  memory allocation issue.  

mFreeMapStack needs to be updated in the case that a return happens, otherwise the global variables will be out of sync.